### ISSUE TITLE
OOM error reported via google play

### ISSUE THREAD

**Issue by ironjan** [ironjan/metal-only#68]
```
java.lang.OutOfMemoryError: 
  at java.util.Arrays.copyOf(Arrays.java:3352)
  at java.lang.AbstractStringBuilder.expandCapacity(AbstractStringBuilder.java:130)
  at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:114)
  at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:606)
  at java.lang.StringBuilder.append(StringBuilder.java:215)
  at com.codingspezis.android.metalonly.player.stream.metadata.IcyStreamMeta.retreiveMetadata(IcyStreamMeta.java:131)
  at com.codingspezis.android.metalonly.player.stream.metadata.IcyStreamMeta.refreshMeta(IcyStreamMeta.java:106)
  at com.codingspezis.android.metalonly.player.stream.metadata.MetadataListener.run(MetadataListener.java:56)
  at java.lang.Thread.run(Thread.java:761)

```

**Comment by ironjan** [comment]
The cause of the issue is in this:
```
while ((c = (char) stream.read()) != -1) {
                strHeaders.append(c);
                if (strHeaders.length() > 5 && (strHeaders.substring((strHeaders.length() - 4), strHeaders.length()).equals("\r\n\r\n"))) {
                    // getEndDate of headers
                    break;
                }
            }
```

AFAIK it is not said that the stream contains the information nor that the stream is finite. Therefore the stream content is copied into a StringBuilder until OOM occurs.

**Comment by ironjan** [comment]
The defect only occurs for Android versions >4.4, mainly for 6.0

Android 6.0	66	81,5 %
Android 5.1	7	8,6 %
Android 4.4	6	7,4 %
Android 5.0	2	2,5 %

**Comment by ironjan** [comment]
Fixed in f8d447773ba1ba546c0e39712f069e3bf09c6200 and 79ad3ebe46cc88d21149b952cff3edcbd2ebe4dd
